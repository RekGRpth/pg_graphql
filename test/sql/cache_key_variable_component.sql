-- Confirm no match returns string, not null
select graphql.cache_key_variable_component('{}') = '';

-- No matches
select graphql.cache_key_variable_component('{}');

select graphql.cache_key_variable_component('{"x": 1}');

select graphql.cache_key_variable_component('{"x": "1"}');

-- Matches
select graphql.cache_key_variable_component('{"id": {"eq": 1}}');

select graphql.cache_key_variable_component('{"orderByVal": "DescNullsFirst"}');

select graphql.cache_key_variable_component('{"orderByObj": [{"email": "AscNullsFirst"}]}');

select graphql.cache_key_variable_component('{
    "id": {"eq": 1},
    "orderByVal": "DescNullsFirst",
    "orderByObj": [
        {"email": "AscNullsFirst"}
    ]
}');
