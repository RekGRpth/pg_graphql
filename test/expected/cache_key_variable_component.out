-- Confirm no match returns string, not null
select graphql.cache_key_variable_component('{}') = '$';
 ?column? 
----------
 t
(1 row)

-- No matches
select graphql.cache_key_variable_component('{}');
 cache_key_variable_component 
------------------------------
 $
(1 row)

select graphql.cache_key_variable_component('{"x": 1}');
 cache_key_variable_component 
------------------------------
 $,$."x"
(1 row)

select graphql.cache_key_variable_component('{"x": "1"}');
 cache_key_variable_component 
------------------------------
 $,$."x"
(1 row)

-- Matches
select graphql.cache_key_variable_component('{"id": {"eq": 1}}');
 cache_key_variable_component 
------------------------------
 $,$."id",$."id"."eq"
(1 row)

select graphql.cache_key_variable_component('{"orderByVal": "DescNullsFirst"}');
           cache_key_variable_component           
--------------------------------------------------
 $,$."orderByVal",$."orderByVal"="DescNullsFirst"
(1 row)

select graphql.cache_key_variable_component('{"orderByObj": [{"email": "AscNullsFirst"}]}');
                                      cache_key_variable_component                                      
--------------------------------------------------------------------------------------------------------
 $,$."orderByObj",$."orderByObj"[0],$."orderByObj"[0]."email",$."orderByObj"[0]."email"="AscNullsFirst"
(1 row)

select graphql.cache_key_variable_component('{
    "id": {"eq": 1},
    "orderByVal": "DescNullsFirst",
    "orderByObj": [
        {"email": "AscNullsFirst"}
    ]
}');
                                                                       cache_key_variable_component                                                                       
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 $,$."id",$."id"."eq",$."orderByObj",$."orderByObj"[0],$."orderByObj"[0]."email",$."orderByObj"[0]."email"="AscNullsFirst",$."orderByVal",$."orderByVal"="DescNullsFirst"
(1 row)

